File: Web/src/api/purchase/purSupplier.ts
import {useBaseApi} from '/@/api/base';

// 供应商主表接口服务
export const usePurSupplierApi = () => {
	const baseApi = useBaseApi("purSupplier");
	return {
		// 分页查询供应商主表
		page: baseApi.page,
		// 查看供应商主表详细
		detail: baseApi.detail,
		// 新增供应商主表
		add: baseApi.add,
		// 更新供应商主表
		update: baseApi.update,
		// 删除供应商主表
		delete: baseApi.delete,
		// 批量删除供应商主表
		batchDelete: baseApi.batchDelete,
		// 导出供应商主表数据
		exportData: baseApi.exportData,
		// 导入供应商主表数据
		importData: baseApi.importData,
		// 下载供应商主表数据导入模板
		downloadTemplate: baseApi.downloadTemplate,
	}
}

// 供应商主表实体
export interface PurSupplier {
	// 主键Id
	id: number;
	// 供应商编码
	supplierCode?: string;
	// 供应商名称
	supplierName?: string;
	// 简称
	shortName: string;
	// 分类ID
	categoryId: number;
	// 联系人
	contactName: string;
	// 联系电话
	phone: string;
	// 手机
	mobile: string;
	// 邮箱
	email: string;
	// 地址
	address: string;
	// 开户银行
	bankName: string;
	// 银行账号
	bankAccount: string;
	// 税号
	taxNo: string;
	// 信用等级(1-5)
	creditRating?: number;
	// 状态(0禁用/1启用)
	status?: number;
	// 是否黑名单
	isBlacklist?: boolean;
	// 备注
	remark: string;
	// 创建时间
	createTime: string;
	// 更新时间
	updateTime: string;
	// 创建者Id
	createUserId: number;
	// 创建者姓名
	createUserName: string;
	// 修改者Id
	updateUserId: number;
	// 修改者姓名
	updateUserName: string;
}

File: CzhERP.Application/Entity/PurSupplier.cs
namespace CzhERP.Application.Entity;

[SugarTable("Pur_Supplier", TableDescription = "供应商主表")]
[SugarIndex("index_{table}_SC", nameof(SupplierCode), OrderByType.Asc, IsUnique = true)]
public class PurSupplier : EntityBase
{
    [SugarColumn(ColumnDescription = "供应商编码", Length = 50, IsNullable = false)]
    public string SupplierCode { get; set; }

    [SugarColumn(ColumnDescription = "供应商名称", Length = 100, IsNullable = false)]
    public string SupplierName { get; set; }

    [SugarColumn(ColumnDescription = "简称", Length = 50)]
    public string? ShortName { get; set; }

    [SugarColumn(ColumnDescription = "分类ID")]
    public long? CategoryId { get; set; }

    [SugarColumn(ColumnDescription = "联系人", Length = 50)]
    public string? ContactName { get; set; }

    [SugarColumn(ColumnDescription = "联系电话", Length = 20)]
    public string? Phone { get; set; }

    [SugarColumn(ColumnDescription = "手机", Length = 20)]
    public string? Mobile { get; set; }

    [SugarColumn(ColumnDescription = "邮箱", Length = 100)]
    public string? Email { get; set; }

    [SugarColumn(ColumnDescription = "地址", Length = 500)]
    public string? Address { get; set; }

    [SugarColumn(ColumnDescription = "开户银行", Length = 100)]
    public string? BankName { get; set; }

    [SugarColumn(ColumnDescription = "银行账号", Length = 50)]
    public string? BankAccount { get; set; }

    [SugarColumn(ColumnDescription = "税号", Length = 50)]
    public string? TaxNo { get; set; }

    [SugarColumn(ColumnDescription = "信用等级(1-5)", DefaultValue = "0")]
    public int CreditRating { get; set; } = 0;

    [SugarColumn(ColumnDescription = "状态(0禁用/1启用)", DefaultValue = "1")]
    public int Status { get; set; } = 1;

    [SugarColumn(ColumnDescription = "是否黑名单", DefaultValue = "false")]
    public bool IsBlacklist { get; set; } = false;

    [SugarColumn(ColumnDescription = "备注", Length = 500)]
    public string? Remark { get; set; }

    [Navigate(NavigateType.OneToOne, nameof(CategoryId))]
    public PurSupplierCategory? Category { get; set; }
}